src/vs/workbench/contrib/terminal/browser/xterm/xtermTerminal.ts

```diff
@@ -768,19 +773,32 @@ class AccessibleBuffer extends DisposableStore {
 
 	private _accessibleBuffer: HTMLElement | undefined;
 	private _bufferElementFragment: DocumentFragment | undefined;
+	private _focusContextKey: IContextKey<boolean>;
 
 	constructor(
 		private readonly _terminal: RawXtermTerminal,
 		private readonly _font: ITerminalFont,
 		private readonly _capabilities: ITerminalCapabilityStore,
 		@IAccessibilityService private readonly _accessibilityService: IAccessibilityService,
-		@IConfigurationService private readonly _configurationService: IConfigurationService
+		@IConfigurationService private readonly _configurationService: IConfigurationService,
+		@IContextKeyService contextKeyService: IContextKeyService
 	) {
 		super();
+		this._focusContextKey = TerminalContextKeys.accessibleBufferFocus.bindTo(contextKeyService);
+		this._accessibleBuffer = this._terminal.element?.querySelector('.xterm-accessibility-buffer') as HTMLElement || undefined;
 		this.add(this._terminal.registerBufferElementProvider({ provideBufferElements: () => this.focus() }));
+		this.add(contextKeyService.onDidChangeContext(e => {
+			if (e.affectsSome(new Set([TerminalContextKeys.focus.key]))) {
+				const terminalTextAreaFocused = contextKeyService.getContextKeyValue(TerminalContextKeys.focus.key);
+				if (terminalTextAreaFocused) {
+					this._focusContextKey.set(false);
+				}
+			}
+		}));
```

Comment:
This will keep accessibleBufferFocus as true if you click in the editor? Can we rely on just blur and focus events handling this?

Also there's a FocusTracker util which can be used for this sort of thing, it handles sub-elements as well: https://github.com/microsoft/vscode/blob/9babe0ed12cf3c7c5918fda8ac378309413bee7a/src/vs/base/browser/dom.ts#L868